src/Covenant/ASG.hs

```diff
@@ -673,6 +669,68 @@ thunk i = do
     ValNodeType t -> throwError . ThunkValType $ t
     ErrorNodeType -> throwError ThunkError
 
+-- | Given a 'Ref' to an algebra (that is, something taking a base functor and
+-- producing some result), and a 'Ref' to a value associated with that base
+-- functor, build a catamorphism to tear it down. This can fail for a range of
+-- reasons:
+--
+-- * First 'Ref' is not a thunk taking one argument
```

Comment:
Could you perhaps rewrite this comment to make it clearer what the arguments *are*, instead of saying what they *are not*? It's a bit hard for me to follow this explanation.